src/Microsoft.Azure.WebJobs.Extensions.Dapr/Services/DaprServiceClient.cs

```diff
@@ -115,12 +145,15 @@ static async Task ThrowIfDaprFailure(HttpResponseMessage response)
                 System.Text.Encoding.UTF8,
                 "application/json");
 
-            HttpResponseMessage response = await this.httpClient.PostAsync(
-                $"{daprAddress}/v1.0/state/{Uri.EscapeDataString(stateStore)}",
-                stringContent,
-                cancellationToken);
+            await DaprHttpCall(async () =>
```

Comment:
Should the response come from `DaprHttpCall`? We can pass function reference and args directly.

Reply:
Same for all other instances.

Reply:
As discussed offline, we can introduce another class `DaprClient` that can be invoked instead of `HttpClient`. The constructor can also take a `daprAddress` parameter. `GetAsync` and `PostAsync` methods on this class will invoke underlying `HttpClient` and look for sidecar not found error.

Reply:
Applied changes in [this PR](https://github.com/Azure/azure-functions-dapr-extension/pull/96/files#diff-b9211a2dbc19fdc2697e06d01e5d60216cd630b7b5b4e845beb1570acc865a21R21)